OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs

```diff
@@ -62,10 +62,26 @@ public FrozenUnderFog(ActorInitializer init, FrozenUnderFogInfo info)
 			startsRevealed = info.StartsRevealed && !init.Contains<ParentActorInit>();
 			var footprintCells = FootprintUtils.Tiles(init.Self).ToList();
 			footprint = footprintCells.SelectMany(c => map.ProjectedCellsCovering(c.ToMPos(map))).ToArray();
+
 			tooltip = Exts.Lazy(() => init.Self.TraitsImplementing<ITooltip>().FirstOrDefault());
 			health = Exts.Lazy(() => init.Self.TraitOrDefault<Health>());
 		}
 
+		public void AddedToWorld(Actor self)
+		{
+			if (initialized)
+				return;
```

Comment:
Perhaps we should get rid of this now?
`AddedToWorld()` will be called only once under normal circumstances. In more special cases, people could end up wanting to change the state of the actor when making it disappear and reappear again (fancy weapons or support powers maybe?).